src/main/java/org/folio/innreach/batch/contribution/service/ContributionJobRunner.java

```diff
@@ -62,86 +61,141 @@ public class ContributionJobRunner {
+
+
+    log.info("Initial: iterationJobId {}", context.getIterationJobId());
+    log.info("Initial: statistics recordsTotal: {}", stats.getRecordsTotal());
+
+    stats.setTopic(topic);
+    stats.setTenantId(context.getTenantId());
 
     var contributionId = context.getContributionId();
-    try (var kafkaReader = itemReaderFactory.createReader(context.getTenantId())) {
-      kafkaReader.open();
-
-      run(context, (centralServerId, stats) -> {
-        while (!isCanceled(contributionId)) {
-          var event = readEvent(kafkaReader);
-          if (event == null) {
-            return;
-          }
-          log.info("Processing instance iteration event = {}", event);
-
-          var instanceId = event.getInstanceId();
-          var iterationJobId = context.getIterationJobId();
-
-          if (isUnknownEvent(event, iterationJobId)) {
-            log.info("Skipping unknown event, current job is {}", iterationJobId);
-            continue;
-          }
-
-          var instance = loadInstanceWithItems(instanceId);
-          if (instance == null) {
-            continue;
-          }
-
-          if (isEligibleForContribution(centralServerId, instance)) {
-            contributeInstance(centralServerId, instance, stats);
-            contributeInstanceItems(centralServerId, instance, stats);
-          } else if (isContributed(centralServerId, instance)) {
-            deContributeInstance(centralServerId, instance, stats);
-          }
-        }
-      });
+
+    var iterationJobId = context.getIterationJobId();
+
+    if (event == null) {
+      log.info("Initial: cannot contribute record for null event, contribution id: {}", contributionId);
+      return;
+    }
+    log.info("Initial: processing instance iteration event: {}", event);
+
+    var instanceId = event.getInstanceId();
+
+    if (isUnknownEvent(event, iterationJobId)) {
+      log.info("Initial: skipping unknown event, current job is {}", iterationJobId);
+      return;
+    }
+
+    Instance instance = loadInstanceWithItems(instanceId);
+
+    if (instance == null) {
+      log.info("Initial: instance is null, skipping");
+      return;
+    }
+
+    var centralServerId = context.getCentralServerId();
+
+    if (isEligibleForContribution(centralServerId, instance)) {
+      contributeInstance(centralServerId, instance, stats);
+      contributeInstanceItems(centralServerId, instance, stats);
+    } else if (isContributed(centralServerId, instance)) {
+      log.info("Initial: deContributeInstance");
+      deContributeInstance(centralServerId, instance, stats);
+    }
+    else {
+      // to test if non-eligible increasing count to verify the stopping condition
+      log.info("Initial: non-eligible instance");
+      ContributionJobRunner.recordsProcessed.put(context.getTenantId(), recordsProcessed.get(context.getTenantId()) == null ? 1
+        : recordsProcessed.get(context.getTenantId())+1);
```

Comment:
This should be using the new `addRecordProcessed` method shouldn't it?

Reply:
addRecordProcessed () is used to restrict onGoing contribution from increasing count and used in methods which is used by both initial and onGoing contribution . In this case not required as this method is only used by initial contribution only.